dotCMS/src/main/java/com/dotmarketing/portlets/htmlpageasset/business/render/HTMLPageAssetRenderedAPIImpl.java

```diff
@@ -256,6 +292,322 @@ public ViewAsPageStatus getViewAsStatus(final HttpServletRequest request,
+//        contextMap.put("EDIT_CONTAINER_PERMISSION" + container.getIdentifier(), hasWritePermissionOnContainer);
+
+        if (Config.getBooleanProperty("SIMPLE_PAGE_CONTENT_PERMISSIONING", true)) {
+
+//            contextMap.put("USE_CONTAINER_PERMISSION" + container.getIdentifier(), true);
+        } else {
+
+//            contextMap.put("USE_CONTAINER_PERMISSION" + container.getIdentifier(), hasReadPermissionOnContainer);
+        }
+    }
+
+    /* Check if we want to accrue the tags associated to each contentlet on
+     * this page
+     */
+//    private void addAccrueTags(final Contentlet contentlet) throws DotDataException {
+//
+//        if (Config.getBooleanProperty("ACCRUE_TAGS_IN_CONTENTS_ON_PAGE", false)) {
+//
+//            // Search for the tags associated to this contentlet inode
+//            final List<Tag> contentletFoundTags = APILocator.getTagAPI().getTagsByInode(contentlet.getInode());
+//            if (contentletFoundTags != null) {
+//                this.pageFoundTags.addAll(contentletFoundTags);
+//            }
+//        }
+//    }
+
+    private boolean needParseContainerPrefix(final Container container, final String uniqueId,
+            final TemplateLayout templateLayout) {
+        String containerIdOrPath = null;
+
+        if (FileAssetContainerUtil.getInstance().isFileAssetContainer(container)) {
+            containerIdOrPath = FileAssetContainerUtil.getInstance().getFullPath((FileAssetContainer) container);
+        } else {
+            containerIdOrPath = container.getIdentifier();
+        }
+
+        return !ParseContainer.isParserContainerUUID(uniqueId) &&
+                (templateLayout == null || !templateLayout.existsContainer(containerIdOrPath, uniqueId));
+    }
+
+    private Contentlet getContentlet(final PersonalizedContentlet personalizedContentlet,
+            final PageMode mode, final long languageId, final User user) {
+
+        return Config.getBooleanProperty("DEFAULT_CONTENT_TO_DEFAULT_LANGUAGE", false)?
+                getContentletOrFallback(personalizedContentlet, mode, languageId, user)
+                : getSpecificContentlet(personalizedContentlet, mode, languageId, user);
+    }
+
+    private Contentlet getSpecificContentlet(final PersonalizedContentlet personalizedContentlet,
+            final PageMode mode, final long requestedLanguage, final User user) {
+        try {
+
+            long languageId = this.resolveLanguageId().isPresent() ? this.resolveLanguageId().get()
+                    : requestedLanguage;
+
+            return APILocator.getContentletAPI().findContentletByIdentifier
+                    (personalizedContentlet.getContentletId(), mode.showLive, languageId,
+                            user, mode.respectAnonPerms);
+        } catch (final DotContentletStateException e) {
+            // Expected behavior, DotContentletState Exception is used for flow control
+            return null;
+        } catch (Exception e) {
```

Comment:
Codacy found an issue: [Avoid catching generic exceptions such as NullPointerException, RuntimeException, Exception in try-catch block](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6090726)